extensions/amp-social-share/1.0/social-share.js

```diff
@@ -60,43 +63,36 @@ export function SocialShare(props) {
   return (
     <div
       role="button"
-      tabindex={props['tabIndex'] || '0'}
+      tabindex={tabIndex || '0'}
       onKeyDown={(e) => handleKeyPress(e, finalEndpoint, checkedTarget)}
       onClick={() => handleActivation(finalEndpoint, checkedTarget)}
-      style={{...size, ...props['style']}}
+      style={{...size, ...style}}
     >
-      {processChildren(props, size)}
+      {processChildren(type, children, color, background, size)}
     </div>
   );
 }
 
 /**
- * If the specified type 'canCustomize' (see config file), allow children
- * to be rendered and color / background to be passed in via props.  If the
- * specified type cannot be customized (canCustomize = false), children
- * will not be rendered and color / background will always be set to default
- * values.
- * @param {!JsonObject} props
+ * If children exist, render the children instead of the icon.  Otherwise,
+ * render the icon associated with the specified type with specified color
+ * and background (or defaults if not specified).
+ * @param {string|undefined} type
+ * @param {?PreactDef.Renderable|undefined} children
+ * @param {string|undefined} color
+ * @param {string|undefined} background
  * @param {JsonObject} size
  * @return {PreactDef.Renderable}
  */
-function processChildren(props, size) {
-  const {
-    'type': type,
-    'children': children,
-    'color': propsColor,
-    'background': propsBackground,
-  } = props;
-
+function processChildren(type, children, color, background, size) {
   if (children) {
     return children;
   } else {
-    const typeConfig = getSocialConfig(type) || dict();
+    const typeConfig = getSocialConfig(/** @type {string} */ (type)) || {};
```

Comment:
If `type` is expected to be a `string`, why not remove `undefined` here and in the type def? That way you won't have to cast it here.

Reply:
Alternatively if it is not expected to be a string, you could give it the empty string as a default value.

Reply:
Yep, so in the `checkProps` function, the `type` will be verified to be a string or it will error out.  So at this point, it can only be a string.  But, since we are still using the original `type` variable with the typedef of `(string|undefined)`, I have to use the cast.  That's my understanding, maybe I am missing your point?

Reply:
If `type` is `string` by the time `processChildren` is called we could update the signature for that at least?

Reply:
Sure, I updated the function signature, and casted type before calling the function in the latest version.